src/bundle/commitments.rs

```diff
@@ -26,84 +25,133 @@ fn hasher(personal: &[u8; 16]) -> State {
+        let mut nh = hasher(ZCASH_ORCHARD_ACTIONS_NONCOMPACT_HASH_PERSONALIZATION);
+
+        for action in bundle.actions().iter() {
+            ch.update(&action.nullifier().to_bytes());
+            ch.update(&action.cmx().to_bytes());
+            ch.update(&action.encrypted_note().epk_bytes);
+            ch.update(
+                &action.encrypted_note().enc_ciphertext.as_ref()
+                    [..Self::OrchardDomain::COMPACT_NOTE_SIZE],
+            );
+
+            mh.update(
+                &action.encrypted_note().enc_ciphertext.as_ref()
+                    [Self::OrchardDomain::COMPACT_NOTE_SIZE
+                        ..Self::OrchardDomain::COMPACT_NOTE_SIZE + MEMO_SIZE],
+            );
+
+            nh.update(&action.cv_net().to_bytes());
+            nh.update(&<[u8; 32]>::from(action.rk()));
+            nh.update(
+                &action.encrypted_note().enc_ciphertext.as_ref()
+                    [Self::OrchardDomain::COMPACT_NOTE_SIZE + MEMO_SIZE..],
+            );
+            nh.update(&action.encrypted_note().out_ciphertext);
+        }
+
+        main_hasher.update(ch.finalize().as_bytes());
+        main_hasher.update(mh.finalize().as_bytes());
+        main_hasher.update(nh.finalize().as_bytes());
+    }
 }
 
 impl OrchardHash for OrchardVanilla {
-    fn update_hash_with_burn(_main_hasher: &mut State, _burn_items: &[(AssetBase, NoteValue)]) {}
+    type OrchardDomain = OrchardVanilla;
+
+    /// Evaluate `orchard_digest` for the bundle as defined in
+    /// [ZIP-244: Transaction Identifier Non-Malleability][zip244]
+    ///
+    /// [zip244]: https://zips.z.cash/zip-0244
+    fn hash_bundle_txid_data<A: Authorization, V: Copy + Into<i64>>(
+        bundle: &Bundle<A, V, OrchardVanilla>,
+    ) -> Blake2bHash {
+        let mut h = hasher(ZCASH_ORCHARD_HASH_PERSONALIZATION);
+
+        Self::update_hash_with_actions(&mut h, bundle);
+
+        h.update(&[bundle.flags().to_byte()]);
+        h.update(&(*bundle.value_balance()).into().to_le_bytes());
+        h.update(&bundle.anchor().to_bytes());
+        h.finalize()
+    }
 }
 
 impl OrchardHash for OrchardZSA {
-    fn update_hash_with_burn(main_hasher: &mut State, burn_items: &[(AssetBase, NoteValue)]) {
+    type OrchardDomain = OrchardZSA;
+
+    /// Evaluate `orchard_digest` for the bundle as defined in
+    /// [ZIP-226: Transfer and Burn of Zcash Shielded Assets][zip226]
+    ///
+    /// [zip226]: https://zips.z.cash/zip-0226
+    fn hash_bundle_txid_data<A: Authorization, V: Copy + Into<i64>>(
+        bundle: &Bundle<A, V, OrchardZSA>,
+    ) -> Blake2bHash {
+        let mut h = hasher(ZCASH_ORCHARD_HASH_PERSONALIZATION);
+        let mut agh = hasher(ZCASH_ORCHARD_ACTION_GROUPS_HASH_PERSONALIZATION);
+
+        Self::update_hash_with_actions(&mut agh, bundle);
+
+        agh.update(&[bundle.flags().to_byte()]);
+        agh.update(&bundle.anchor().to_bytes());
+        agh.update(&[0, 0, 0, 0]); // timeLimit is always equal to 0
```

Comment:
why 4 zeros?

Reply:
I see.
I think we should add `timelimit` to the bundle and serialize it. Basically treat it as a real bundle field. 
For NU7, `timelimit` is locked to `0x00000000` by the consensus. But the lock will be removed in future versions.

Reply:
@vivek-arte why 4 bytes? `timelimit` is a block number, what is the datatype zcash use for block numbers? what is the maximum possible block number?

Reply:
Actually missed this, I actually set `timeLimit` to be of type `uint32` in ZIP 230 (see [rendered version of zips/#78](https://deploy-preview-78--zcash-zips-qedit.netlify.app/zip-0230#orchardzsa-action-group-description)). That is encoded as 4 bytes, hence the mention of `(4 bytes)` in `T.4a.vi`.

The reasoning is, I followed the idea for the `nExpiryHeight` field higher up in the transaction, that is also of type `uint32`.  `nExpiryHeight` is the height after which the transaction is removed from mempools if not mined, so it stands to reason that `timeLimit` shouldn't exceed the maximum value for `nExpiryHeight`
![image](https://github.com/user-attachments/assets/3e9a333d-ed5b-49ee-a543-1b68a44fba9e)

To quote from ZIP 203 (why that is a reasonable maximum):
> Maximum: 499999999, which is about 1185 years from now at 75 seconds/block.

Reply:
@ConstanceBeguier 
To summarize, we renamed `timeLimit` to `nAGExpiryHeight`, and it is of type `uint32` in the spec. (so `u32` here).

https://deploy-preview-78--zcash-zips-qedit.netlify.app/zip-0230#orchardzsa-action-group-description

Reply:
I added the field expiry_height in the Bundle structure.
Now the SIGHAH is evaluated with this field and no longer a hard coded 0 value for expiry height